### PR TITLE
added the postgres `cardinality` function

### PR DESCRIPTION
**Pull request by aznszn**
added the postgres `cardinality` function under [#4171 ](https://github.com/diesel-rs/diesel/issues/4153)

### CODE REVIEW

**Review by guissalustiano — APPROVED** [review]
Thanks for opening the PR!
The changes looks good to me, just a minor style suggestion that can be ignored

**Review by guissalustiano — APPROVED** [review]

File: diesel/src/pg/expression/functions.rs [hunk 1]
```diff
@@ -793,3 +793,41 @@ define_sql_function! {
...
+    ///     .get_result::<Option<i32>>(connection)?;
+    /// assert_eq!(0, array_cardinality.unwrap());
+    /// 
+    /// let array_cardinality = diesel::select(cardinality::<Nullable<Array<Integer>>, _>(None::<Vec<i32>>))
+    ///     .get_result::<Option<i32>>(connection)?;
+    /// assert_eq!(None, array_cardinality);
+    /// # Ok(())
+    /// # }
+    ///
+    fn cardinality<Arr:ArrayOrNullableArray<> + SingleValue>(a: Arr) -> Nullable<Integer>;
```
Comment (guissalustiano) [inline diesel/src/pg/expression/functions.rs:832]: I prefer without the empty brackets here to follow the same pattern from others associate types uses
```suggestion
    fn cardinality<Arr:ArrayOrNullableArray + SingleValue>(a: Arr) -> Nullable<Integer>;
```

**Review by guissalustiano — APPROVED** [review]
Thanks for opening the PR!
The changes looks good to me, I have just a minor style suggestion.

**Comment by guissalustiano** [comment]
Oops, I though I could delete my review that I forgot to link, sorry for the spam 😅

**Comment by aznszn** [comment]
> Oops, I though I could delete my review that I forgot to link, sorry for the spam 😅

its cool no worries

**Comment by aznszn** [comment]
@guissalustiano can you check why the test is failing after I committed the suggest change

**Review by weiznich — APPROVED** [review]
Thanks for working on this :+1:

I just rebased the PR and fixed a the documentation

**Pull request merged**